src/dotnet/commands/RestoringCommand.cs

```diff
@@ -20,7 +20,8 @@ private IEnumerable<string> ArgsToForwardToRestore()
                 !a.StartsWith("/t:") &&
                 !a.StartsWith("/target:") &&
                 !a.StartsWith("/ConsoleLoggerParameters:") &&
-                !a.StartsWith("/clp:"));
+                !a.StartsWith("/clp:") &&
+                !a.StartsWith("/p:TargetFramework"));
```

Comment:
Is this iterating over all msbuild arguments (including user arguments) or just the arguments that the CLI added on user behalf (e.g. -f --> /p:TargetFramework, etc.)?

If the former, then we'll get different behavior with different user syntax. If the latter, it's OK, but why the duplication? This deserves a comment at a minimum.